hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/cdc/InternalRowToJsonStringConverter.scala

```diff
@@ -36,15 +37,63 @@ class InternalRowToJsonStringConverter(schema: StructType) {
   }
 
   def convert(record: InternalRow): UTF8String = {
-    val map = scala.collection.mutable.Map.empty[String, Any]
+    // Use LinkedHashMap to preserve field order
+    val map = scala.collection.mutable.LinkedHashMap.empty[String, Any]
     schema.zipWithIndex.foreach {
       case (field, idx) =>
-        if (field.dataType.isInstanceOf[StringType]) {
-          map(field.name) = Option(record.getUTF8String(idx)).map(_.toString).orNull
```

Comment:
Have you run a functional test on Spark with CDC to verify that `StringType` column still works?